hawkular-command-gateway-api/src/main/resources/schema/A0uthMessage.schema.json

```diff
@@ -25,6 +49,27 @@
   "properties": {
     "authentication": {
       "$ref": "#/definitions/authentication"
+    },
+    "recipientType": {
+      "$ref": "#/definitions/recipientType"
+    },
+    "recipient": {
+      "description" : "feedId or uiId",
+      "type": "string"
+    },
+    "senderType": {
+      "$ref": "#/definitions/recipientType"
+    },
+    "sender": {
+      "description" : "feedId or uiId",
+      "type": "string"
+    },
+    "attachments": {
+      "description" : "A list of binary attachments if any.",
+      "type": "array",
+      "items" : {
+        "$ref": "#/definitions/attachment"
+      }
```

Comment:
I don't like having this in an "AuthMessage" since this attachments thing doesn't have anything to do with authentication.

Also, keep in mind that the cmdgw server will scrub the entire Authentication portion of the message as soon as the message is received (we actually null it out). That would mean once we authenticate the message, everything in AuthMessage will go away - so we will lose all information about attachments, too. Which is why we should not have anything in AuthMessage that isn't related to authentication. And we need to keep in mind that everything in Auth Message will be immediately removed once it is received by the cmdgw server.